sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1066,123 @@ def _eval_is_nonnegative(self):
+
+    .. math:: \frac{(k_1+k_2+\cdots+k_m)!}{k_1!\, k_2! \cdots k_m!} = {k_1\choose k_1}{k_1+k_2\choose k_2}\cdots{k_1+k_2+\cdots+k_m\choose k_m}
+
+    The multinomial coefficients have a direct combinatorial interpretation,
+    as the number of ways of depositing n distinct objects into m distinct bins,
+    with k_1 objects in the first bin, k_2 objects in the second bin, and so on.
+
+    To interpret using non-integral values of `k` one can consider an expansion series of power `n`,
+    `(a_1 + a_2 + ... + a_m) ^ n` such that `(k_1 + k_2 + ... + k_m) = n` where `k_i` is the
+    exponent of `a_i` after expansion.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Rational, multinomial, expand_func, symbols, gamma
+    >>> n = Symbol('n', integer=True, positive=True)
+
+    >>> multinomial(15, 8)
+    490314
+
+    Comparing coefficents generated with the multinomial function:
+
+    >>> a, b, c = symbols('a b c')
+    >>> ((a+b+c)**3).expand()
+    a**3 + 3*a**2*b + 3*a**2*c + 3*a*b**2 + 6*a*b*c + 3*a*c**2 + b**3 + 3*b**2*c + 3*b*c**2 + c**3
+
+    The coefficient for any term with variables having exponents of 0, 1 and 2 is 3:
+
+    >>> multinomial(2, 1, 0)
+    3
+
+    A series `(1 + x)^n`, where `n` is 2/3 when expanded, would yield
+    `1 + (2/3) x^1 - (1/9) x^2 + (4/81) x^3 - (7/243) x^4 + ...`.
+    To find the coeffient of `x^1`, one needs to use
+    `multinomial(1, a)` where `1 + a = n; a = -1/3`.
+
+    >>> multinomial(1, Rational(-1, 3))
+    2/3
+
+    This is equal to 2/3 which was the coefficient of `x^1`.
+
+    >>> multinomial(n, 3).rewrite(gamma)
+    gamma(n + 4)/(6*gamma(n + 1))
+
+    >>> multinomial(n, 3).expand(func=True)
+    n**3/6 + n**2 + 11*n/6 + 1
+
+    >>> expand_func(multinomial(n, 3))
+    (n + 1)*(n + 2)*(n + 3)/6
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Multinomial_theorem#Multinomial_coefficients
+
+    .. [2] http://functions.wolfram.com/GammaBetaErf/Multinomial/02/0001/
+
+    """
+
+    @classmethod
+    def eval(self, *args):
+        args = list(map(sympify, args))
+        if len(args) < 1:
+            raise ValueError("A miniumum of 1 arg required")
+        if any(x.is_integer and x.is_negative for x in args):
+            return S.Zero
+        else:
+            rv = self._binomial(*args)
+            if rv is not None and not rv.has(binomial):
+                return rv
+
+    def _eval_evalf(self, prec):
+        rv = self._binomial(*self.args)
+        if rv is not None:
+            return rv._evalf(prec)
+
+    @classmethod
+    def _binomial(self, *args):
+            from sympy import binomial
+            def runsum(a):
+                t = 0
+                for i in a:
+                    t += i
+                    yield t
+            rv = Mul(*[
+                binomial(j, k) for j, k in zip(runsum(args), args)])
+            if any(x.is_Float for x in args):
+                rv = rv.expand()
+            return rv
```

Comment:
You should dedent these lines

Reply:
I added the `if` for which these were indented.